src/main/java/bio/terra/service/dataset/flight/create/CreateDatasetAuthzIamStep.java

```diff
@@ -33,9 +33,11 @@ public CreateDatasetAuthzIamStep(
     public StepResult doStep(FlightContext context) throws InterruptedException {
         FlightMap workingMap = context.getWorkingMap();
         UUID datasetId = workingMap.get(DatasetWorkingMapKeys.DATASET_ID, UUID.class);
-        Map<IamRole, String> policyEmails = iamClient.createDatasetResource(userReq, datasetId);
+        iamClient.createDatasetResource(userReq, datasetId);
+        Map<IamRole, String> policyEmails = iamClient.syncDatasetResourcePolicies(userReq, datasetId);
```

Comment:
I'm probably missing something, but I don't see why there needs to be two entry points for these two operations. If we are always doing `syncDatasetResourcePolicies()` after `createDatasetResource()`, can this be done inside `createDatasetResource()`? There can still be two different retries internally.

Reply:
You're right! It was a last minute change to pull them back into the same stairway step. But, I could combine them at a lower level, in SamIam.